diff --git a/thirdparty/thorvg/src/renderer/tvgPaint.h b/thirdparty/thorvg/src/renderer/tvgPaint.h
index 9ae2fb11eb..0f60b846af 100644
--- a/thirdparty/thorvg/src/renderer/tvgPaint.h
+++ b/thirdparty/thorvg/src/renderer/tvgPaint.h
@@ -122,18 +122,18 @@ namespace tvg
             return tr.m;
         }
 
-        void clip(Paint* clipper)
+        void clip(Paint* clp)
         {
             if (this->clipper) {
                 P(this->clipper)->unref();
-                if (this->clipper != clipper && P(this->clipper)->refCnt == 0) {
+                if (this->clipper != clp && P(this->clipper)->refCnt == 0) {
                     delete(this->clipper);
                 }
             }
-            this->clipper = clipper;
-            if (!clipper) return;
+            this->clipper = clp;
+            if (!clp) return;
 
-            P(clipper)->ref();
+            P(clp)->ref();
         }
 
         bool composite(Paint* source, Paint* target, CompositeMethod method)
